File: Command Line Software/Packets/ConnectionConfirmPacket.py
class ConnectionConfirmPacket:
    # Size of the packet in bytes
    PACKET_SIZE = 3

    @staticmethod
    def get_packet_values(packet_bytes : bytearray):
        conn_estab : int = packet_bytes[0]
        error : int = packet_bytes[1]
        setup : int = packet_bytes[2]

        return (conn_estab, error, setup)

    

File: smart_plug/connection_confirm_packet.c
#include <stdint.h>
#include <stdlib.h>

#include "connection_confirm_packet.h"

void create_connection_packet(Conn_Conf_Pack_t* packet, uint8_t* packet_buf)
{
    packet_buf[0] = packet->conn_estab;
    packet_buf[1] = packet->error;
    packet_buf[2] = packet->setup;
}